### ISSUE TITLE
New `now` function

### ISSUE THREAD

**Issue by binarylogic** [timberio/vector#3753]
The `now` function generates a new timestamp for the current time in UTC.

## Examples

```
.timestamp = now()
```

**Comment by JeanMertz** [comment]
@binarylogic while working on this, I was wondering what your use-case is for allowing a time zone to be specified.

Is it only for you to be able to determine which offset is added to the timestamp itself? If so, I'm not sure what value that adds. It would be simpler (in a usability sense) to just provide a `now()` function that returns the current time in UTC. Any system ingesting this time will be able to shift the time with the correct offset accordingly.

If you think there is a use-case for wanting the timestamp to actually be shown in a specific time zone, then it might make more sense to have a separate function that takes a timestamp, and shifts the time zone. This would allow other timestamp related functions (such as `to_timestamp` and `parse_timestamp`) to also be chained (cc #3736) with the hypothetical `timestamp_with_offset` function.

Basically, what I'm suggesting is, let `now()` return (e.g.) `"2020-09-15T10:38:34Z"` and then (optionally, if you still think there's a use-case for this) have `timestamp_with_offset(.timestamp, "US/Eastern")` print `"2020-09-15T05:38:34-05:00"`.

**Comment by JeanMertz** [comment]
Another option (currently not possible, but worth considering) is to not have `timestamp_with_offset` at all, and just have all time-based functions accept an optional `offset` named argument (cc #3851), e.g.:

* `now()`
* `now(offset = "US/Eastern")`
* `to_timestamp(.foo, offset = "US/Eastern", default = "2020-09-15T10:38:34Z")`
* `to_timestamp(.foo, offset = -5)`
* `parse_timestamp(.foo, "%s", with_timezone = "US/Eastern", default = "2020-09-15T10:38:34Z")`

   _side note: with optional named arguments, `parse_timestamp` feels redundant. We could just add a `format` argument to `to_timestamp`, which forces the function to use that format explicitly, or error if the string can't be parsed, whereas without the argument, it would try the set of formats we currently have hardcoded (including UNIX and RFC3339):_

   * `to_timestamp(.foo, format = "%s", offset = "US/Eastern", default = "2020-09-15T10:38:34Z")`

**Comment by binarylogic** [comment]
@JeanMertz that's my fault. Let's ignore time zones for now since we don't support them in our data model anyway. I've updated the description to return UTC only.
